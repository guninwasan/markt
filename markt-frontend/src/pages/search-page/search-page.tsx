import { ExploreListings, Navbar, SearchResults } from "../../components";

const SearchPage = () => {
  return (
    <>
      <Navbar />
      <SearchResults />
      <ExploreListings />
    </>
  );
};

export { SearchPage };
