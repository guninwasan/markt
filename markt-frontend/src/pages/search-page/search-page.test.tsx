import React from "react";
import { render } from "@testing-library/react";
import { SearchPage } from "./search-page";

jest.mock("../../components", () => ({
  ExploreListings: () => <div>ExploreListings Mock</div>,
  Navbar: () => <div>Navbar Mock</div>,
  SearchResults: () => <div>SearchResults Mock</div>,
}));

describe("SearchPage", () => {
  it("renders the Navbar, SearchResults, and ExploreListings components", () => {
    const { getByText } = render(<SearchPage />);

    expect(getByText("Navbar Mock")).toBeInTheDocument();
    expect(getByText("SearchResults Mock")).toBeInTheDocument();
    expect(getByText("ExploreListings Mock")).toBeInTheDocument();
  });

  it("should match the snapshot", () => {
    const { container } = render(<SearchPage />);
    expect(container).toMatchSnapshot();
  });
});
