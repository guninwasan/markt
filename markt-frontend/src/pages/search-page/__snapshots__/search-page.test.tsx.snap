// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchPage should match the snapshot 1`] = `
<div>
  <div>
    Navbar Mock
  </div>
  <div>
    SearchResults Mock
  </div>
  <div>
    ExploreListings Mock
  </div>
</div>
`;
